core/domain/topic_jobs_one_off.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for topics."""
+
+import logging
+
+from core import jobs
+from core.domain import topic_domain
+from core.domain import topic_services
+from core.platform import models
+import feconf
+
+(topic_models,) = models.Registry.import_models([models.NAMES.topic])
+
+
+class TopicMigrationJob(jobs.BaseMapReduceOneOffJobManager):
```

Comment:
Here and elsewhere, let's maybe name these to end with *OneOffJob, for clarity.

Also, I noticed that these jobs don't seem to be in the job registry. Did you try running them manually?

Reply:
> Here and elsewhere, let's maybe name these to end with *OneOffJob, for clarity.

Sure, but that would break convention from `collection_jobs_one_off.CollectionMigrationJob` and `exp_jobs_one_off.ImageDataMigrationJob`. Should they be changed as well?

> Also, I noticed that these jobs don't seem to be in the job registry. Did you try running them manually?

Yes, but I omitted putting the jobs in the job registry before there are multiple versions of models. As far as I understand, they are unreleased so it doesn't make sense to have these in the job registry right now? Or should they be added regardless?

Reply:
Oh, that's a good point. I think, let's put them in if running them is a no-op, but guard the map() with a check on feconf.ENABLE_NEW_STRUCTURES. Then, when that flag is removed, everything will behave correctly post-removal.

Reply:
> Oh, that's a good point. I think, let's put them in if running them is a no-op, but guard the map() with a check on feconf.ENABLE_NEW_STRUCTURES. Then, when that flag is removed, everything will behave correctly post-removal.

fixed

> Sure, but that would break convention from collection_jobs_one_off.CollectionMigrationJob and exp_jobs_one_off.ImageDataMigrationJob. Should they be changed as well?

can you confirm this as well?

Reply:
renamed.